src/components/tables/TableConnected.tsx

```diff
@@ -0,0 +1,79 @@
+import { TableChildComponent } from './TableConstants';
+import { getTableChildComponentId } from './TableUtils';
+import { ITableOwnProps, ITableProps, Table } from './Table';
+import { ITableState } from './TableReducers';
+import { addActionsToActionBar } from '../actions/ActionBarActions';
+import { addTable, removeTable } from './TableActions';
+import { IReactVaporState, IReduxActionsPayload } from '../../ReactVapor';
+import { IReduxAction, ReduxUtils } from '../../utils/ReduxUtils';
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { ThunkAction } from 'redux-thunk';
+import * as _ from 'underscore';
+import { IActionOptions } from '../actions/Action';
+import {
+  defaultTableStateModifyerThunk,
+  serverTableStateModifyerThunk,
+} from './TableThunkActionCreators';
+import { IDropdownOption } from '../dropdownSearch/DropdownSearch';
+import { selectOptionDropdownSearch, closeDropdownSearch } from '../dropdownSearch/DropdownSearchActions';
+import { PER_PAGE_NUMBERS } from '../navigation/perPage/NavigationPerPage';
+
+export interface ITableDispatchProps {
+  onDidMount: () => void;
+  onUnmount: () => void;
+  onModifyData: (shouldResetPage: boolean) => void;
+  onRowClick: (actions: IActionOptions[]) => void;
+  onPredicateOptionClick: (predicateId: string, option: IDropdownOption) => void;
+}
+
+const mapStateToProps = (state: IReactVaporState, ownProps: ITableOwnProps) => {
+  const tableState: ITableState = state.tables[ownProps.id];
+
+  return { tableState: { ...tableState } };
```

Comment:
You can simply return the state without having to copy it into a new object. The reducers will make sure the state is immutable on the other end.

Reply:
thanks